homeassistant/components/kostal_plenticore/sensor.py

```diff
@@ -98,6 +105,40 @@ async def async_setup_entry(
             )
         )
 
+    # Accumulated PV-DC power: The inverter does not provide a sum
+    # of all power of the PV strings. Hence a PlenticoreComputedSensor
+    # is used to provide this information from each existing PV string.
+    dc_inputs = list(
+        filter(
+            lambda module_data_id: module_data_id[0] in available_process_data
+            and module_data_id[1] in available_process_data[module_data_id[0]],
+            [
+                ("devices:local:pv1", "P"),
+                ("devices:local:pv2", "P"),
+                ("devices:local:pv3", "P"),
+            ],
+        )
+    )
+    if dc_inputs:
+        entities.append(
+            PlenticoreComputedSensor(
+                process_data_update_coordinator,
+                entry.entry_id,
+                entry.title,
+                dc_inputs,
+                "PV-DC Power",
+                {
+                    ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
+                    ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+                },
+                plenticore.device_info,
+                lambda powers: PlenticoreDataFormatter.format_round(
+                    sum(float(power) for power in powers)
```

Comment:
Integrations that integrate devices or services should not compute new state. They should just provide the state of the device or service. We have other integrations that compute state, eg template integrations.

Reply:
So, does that mean this PR has no chance to get integrated?

Reply:
Yes, sorry. We divide the responsibilities between integrations that integrate devices or services and integrations that calculate state based of input from other integrations or user templates. We don't allow mixing this up.